src/Voron/StorageEnvironment.cs

```diff
@@ -564,7 +564,8 @@ internal LowLevelTransaction NewLowLevelTransaction(TransactionPersistentContext
                 }
                 finally
                 {
-                    _envDispose.Signal();
+                    if (_envDispose.IsSet == false)
```

Comment:
Oh, so we have a problem if we try to signal the event after ti was already disposed?